src/bundle.rs

```diff
@@ -242,13 +242,28 @@ impl<T: Authorization> Bundle<T> {
         todo!()
     }
 
+    /// Construct a new bundle by applying a transformation that might fail
+    /// to the value balance.
+    pub fn try_map_value_balance<V0, E, F: FnOnce(V) -> Result<V0, E>>(
```

Comment:
In what context would this API be used? I can't see any need for it during parsing.

Reply:
Right now it's just being used for testing, but in general if a type can be made traversable over one of its type parameters, it's useful to do so. It's a shame that traverse is limited to `Result` of course, but it's a good enough approximation.

Reply:
I would prefer that we keep this crate-internal until there is a concrete external use for it. It's easy to expose it later if necessary.

Reply:
What I mean is, I currently use it in the test code in librustzcash.